### PR TITLE
Add restriction ttl

### PR DESCRIPTION
**Pull request by dlongley**

### CODE REVIEW

**Comment by dlongley** [comment]
Closes #28.

File: lib/ResourceTokenizer.js [hunk 1]
```diff
@@ -177,18 +178,36 @@ export class ResourceTokenizer {
...
+    /* Compute the new TTL for the acquisition record by considering the
+    previous value and the maximum TTL from the applied restrictions. The new
+    TTL must be greater than or equal to the previous one unless it has been
+    set to `0` to indicate there are no current acquisitions. This ensures that
+    previous resource acquisitions continue to be tracked for at least as long
+    as was required when the resources were first acquired. */
+    const {maxRestrictionTtl} = checkResults;
+    const {previousAcquisitionTtl: previousTtl} = this;
+    // new TTL must be no less than the previous TTL (unless it is zero)
+    const ttl = Math.max(previousTtl || maxRestrictionTtl, maxRestrictionTtl);
```
Comment (mattcollier) [inline lib/ResourceTokenizer.js:201]: is it not adequate to just do?
```suggestion
    const ttl = Math.max(previousTtl, maxRestrictionTtl);
```

File: lib/resources.js [hunk 2]
```diff
@@ -311,18 +318,21 @@ async function _getAcquisitionRecord({acquirerId}) {
...
+        }],
+        expires: new Date(),
+        ttl: 0
       }
     };
   }
+  // expose `expires` as milliseconds since the epoch
   if(record.acquisition.expires) {
     record.acquisition.expires = record.acquisition.expires.getTime();
   }
```
Comment (mattcollier) [inline lib/resources.js:330]: Is this conditional still required?  I see expires is now part of the default record.  Is there a case when `record.acquisition.expires` is falsey?

File: lib/resources.js [hunk 3]
```diff
@@ -311,18 +318,21 @@ async function _getAcquisitionRecord({acquirerId}) {
         tokenized: [{
           tokenizerId,
           resources: {}
-        }]
+        }],
+        expires: new Date(),
```
Comment (mattcollier) [inline lib/resources.js:322]: I'm confused about this.  Why is this being sent to now?  Seems to me anyone looking at this record downstream is going to think it's expired already.  Am I misreading this?

File: test/mocha/20-resources.js [hunk 4]
```diff
@@ -553,4 +554,84 @@ describe('resources', function() {
...
+          resource: RESOURCES.PLUM,
+          count: 1
+        }],
+        untrackedResources: []
+      };
+      assertCheckResult(result, expectedResult);
+    }
+
+    // wait for expiration period
+    await new Promise(r => setTimeout(r, seconds * 1000));
```
Comment (mattcollier) [inline test/mocha/20-resources.js:617]: please use `bedrock.util.delay`.
```suggestion
    await delay(seconds * 1000);
```
https://www.npmjs.com/package/delay

**Review by mattcollier — CHANGES_REQUESTED** [review]

File: lib/ResourceTokenizer.js (hunk 1 above)
Comment (dlongley) [reply lib/ResourceTokenizer.js:201]: At this point, I think that's fine. Previously there was a chance that `previousTtl` would be undefined. Will update.

File: lib/resources.js (hunk 2 above)
Comment (dlongley) [reply lib/resources.js:330]: This was before the decision to require the collection to be dropped. I'll remove the conditional now.

File: lib/resources.js (hunk 3 above)
Comment (dlongley) [reply lib/resources.js:322]: No, you're not misreading it. This is the default record -- and it should expire immediately if nothing gets added to it. Setting it to any other time doesn't make any sense, would be arbitrary, and could cause it to unnecessarily persist.

File: test/mocha/20-resources.js (hunk 4 above)
Comment (dlongley) [reply test/mocha/20-resources.js]: Updated.

File: lib/resources.js (hunk 3 above)
Comment (mattcollier) [reply lib/resources.js:322]: @dlongley you've done a great job documenting other aspects of this code.  If some commentary could be added here as well, that would be great.

Comment (dlongley) [reply lib/resources.js:325]: Added comments.

**Pull request merged**